src/main/java/com/rtb/projectmanagementtool/user/UserProfileServlet.java

```diff
@@ -109,24 +108,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
   public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws ServletException, IOException {
 
-    Long userID = auth.whichUserIsLoggedIn(request, response);
+    // get parameters
+    long taskID = Long.parseLong(request.getParameter("taskID"));
+    String message = request.getParameter("message");
 
-    // get this user's tasks
-    TaskController taskController = new TaskController(datastore);
-    ArrayList<TaskData> tasks = taskController.getTasksByUserID(userID);
+    // new controller for methods
+    PrivateCommentController pcController = new PrivateCommentController(datastore);
 
-    // for each task, get params and create private comments
-    for (TaskData task : tasks) {
-      long taskID = Long.parseLong(request.getParameter("taskID"));
-      String message = request.getParameter("message");
+    // if a pcomment already exists with this taskID. Get it and update it.
+    if (pcController.getPrivateCommentByTaskID(taskID) != null) {
+      PrivateCommentData pcData = pcController.getPrivateCommentByTaskID(taskID);
+      // update message
+      pcData.setMessage(message);
 
-      // Create PrivateCommentData object
-      PrivateCommentData privateComment = new PrivateCommentData(taskID, userID, message);
+      // Add updates to datastore
+      pcController.updatePrivateComment(pcData);
+    }
 
-      // Add private comment to datastore
-      PrivateCommentController privateCommentController = new PrivateCommentController(datastore);
-      privateCommentController.addPrivateComment(privateComment);
+    // otherwise, create a new pcData and put that in ds
```

Comment:
Update: it seems this is being hit every time

Reply:
```
long taskID = Long.parseLong(request.getParameter("taskID-" + task.getName()));
```
taskID is probably null because that specific parameter doesn't exist. In jsp you send in 
```
name="taskID-"<%=task.getName()%>
```
Only the 'taskID-' part is sent to the servlet. Putting <%=task.getName()%> inside the quotation mark should do:
```
name="taskID-<%=task.getName()%>"
```

Reply:
If this is getting hit every time, it means `pcController.getPrivateCommentByTaskID(taskID)` either is not finding the record (check the query and filters), or it's not filling in the key and/or taskID correctly (so looking up the taskID is not returning results)